src/expr/src/relation/mod.rs

```diff
@@ -2769,6 +2769,18 @@ impl RowSetFinishing {
             })
             .unwrap_or(limit);
 
+        // Bail early if we know our results won't fit into our Vec
+        //
+        // Note: this check is a _lower bound_ as to how much memory
+        // we'll need for results. Calculating the actual answer would
+        // require us to iterate through all of the rows
```

Comment:
Wait, we're actually iterating through all the rows in the `try_fold` above!

Reply:
ah ha! updated!